contributingGuides/philosophies/PAGINATION.md

```diff
@@ -0,0 +1,113 @@
+    2. Post-process network responses using the [Pagination middleware](https://github.com/Expensify/App/blob/1a06fa4add10b53a1a9266927d3b08a4ca35d3c4/src/libs/Middleware/Pagination.ts) to keep track of the start and end point of the page you loaded in a request, and [merge it with existing pages if it overlaps](https://github.com/Expensify/App/blob/1a06fa4add10b53a1a9266927d3b08a4ca35d3c4/src/libs/PaginationUtils.ts#L104).
+    3. When rendering your list, use that pages key for your sorting order, and insert "gap markers" between the edges of the pages you've loaded.
+    4. When rendering your list, [only render a single continuous chunk](https://github.com/Expensify/App/blob/1a06fa4add10b53a1a9266927d3b08a4ca35d3c4/src/libs/PaginationUtils.ts#L166) containing your current "anchor point" (the reportAction you linked to, for example), up until you reach the end of the list in either direction or a gap marker.
+    5. Then when you scroll, you'll hit your gap marker and can make network requests to fill in the gap.
+
+    More details can be found in [the Pagination middleware](https://github.com/Expensify/App/blob/1a06fa4add10b53a1a9266927d3b08a4ca35d3c4/src/libs/Middleware/Pagination.ts). Efforts were made to generalize this code, but so far it has only been used for reportActions.
+
+## Future improvements
+The following ideals have not (yet) been implemented in Expensify, but we'll hold them out here as aspirational goals for future pagination systems, inspired by WhatsApp.
+
+### Layered Pagination
+The pagination systems we've described so far paginate data from the client to the server. i.e: client loads one page, then when they reach the edge of the page they ask the server for the next page. This goes a long way to decouple the total volume of data a user has access to from their app's performance.
+
+However, the systems we've built and/or described have no data eviction. Once data has been downloaded from the server, it's saved to disk, and because of the design of Onyx all that data is typically kept in RAM in the Onyx cache. If this starts to become a problem ([it has been discussed before](https://expensify.slack.com/archives/C05LX9D6E07/p1755077335799619)), throwing away data on disk is not the best approach. Instead, we should build a two-layer pagination system:
+
+1. Data is loaded into RAM one page at a time.
+2. If we reach the edge of the page, we first check the disk for more.
+3. Only if the page we requested is not on disk, we request it from the server.
+4. If data must be evicted due to slow performance, it is first evicted from RAM.
+5. Data should be evicted from disk only if reading data from disk becomes a bottleneck, or we reach device storage limits.
+
+That way, we can preserve our [Offline Philosophy](https://github.com/Expensify/App/blob/main/contributingGuides/philosophies/OFFLINE.md) and provide a first-class offline UX without compromising performance.
```

Comment:
FYI these links can be shorted to 
```suggestion
That way, we can preserve our [Offline Philosophy](/contributingGuides/philosophies/OFFLINE.md) and provide a first-class offline UX without compromising performance.
```

Which is better if the repo is ever renamed